homeassistant/components/ezviz/camera.py

```diff
@@ -32,166 +50,194 @@
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_CAMERAS, default={}): {cv.string: CAMERA_SCHEMA},
+        vol.Optional(CONF_REGION, default=DEFAULT_REGION): cv.string,
+        vol.Optional(ATTR_CAMERAS, default={}): {cv.string: CAMERA_SCHEMA},
     }
 )
 
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_SESSION_RENEW = timedelta(seconds=90)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Ezviz IP Cameras."""
 
-    conf_cameras = config[CONF_CAMERAS]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up a Ezviz IP Camera from platform config."""
+    _LOGGER.warning(
+        "Loading ezviz via platform config is deprecated, it will be automatically imported. Please remove it afterwards."
+    )
 
-    account = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
+    if ATTR_CAMERAS in config:
+        cameras_conf = config.get(ATTR_CAMERAS, CAMERA_SCHEMA)
 
-    try:
-        ezviz_client = EzvizClient(account, password)
-        ezviz_client.login()
-        cameras = ezviz_client.load_cameras()
+        _LOGGER.debug("camera constant: %s", cameras_conf)
+        for camera in cameras_conf.items():
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_IMPORT},
+                    data=camera,
+                )
+            )
 
-    except PyEzvizError as exp:
-        _LOGGER.error(exp)
-        return
+    _LOGGER.debug("full config: %s", config)
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-    # now, let's build the HASS devices
-    camera_entities = []
 
-    # Add the cameras as devices in HASS
-    for camera in cameras:
+async def async_setup_entry(
+    hass, entry, async_add_entities: Callable[[List[Entity], bool], None]
+) -> None:
+    """Set up Ezviz cameras based on a config entry."""
 
-        camera_username = DEFAULT_CAMERA_USERNAME
-        camera_password = ""
-        camera_rtsp_stream = ""
-        camera_serial = camera["serial"]
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    ffmpeg_arguments = entry.options.get(
+        CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+    )
+    camera_entities = []
+
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
```

Comment:
Please don't add any new features that aren't strictly needed to retain the existing feature set. New features can be added in follow up PRs. This PR should just migrate the integration to use config entries with as little changes as possible.

Reply:
Noted and Thank you for all your patience.

Reply:
Please remove the custom services for now and move that addition to a follow up PR.

Reply:
Removed all custom services.